AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/CreateKeys.dfy

```diff
@@ -330,6 +349,27 @@ module {:options "/functionSyntax:4" } CreateKeys {
     var activeBranchKeyContext := Structure.ActiveBranchKeyEncryptionContext(decryptOnlyBranchKeyContext);
     var beaconBranchKeyContext := Structure.BeaconKeyEncryptionContext(decryptOnlyBranchKeyContext);
 
+    if !HvUtils.HasUniqueTransformedKeys?(activeBranchKeyContext) {
+      return Failure(Types.BranchKeyCiphertextException(
+                       message := ErrorMessages.NOT_UNIQUE_BRANCH_KEY_CONTEXT_KEYS
+                     ));
+    }
+    var ecToKMS := HvUtils.SelectKmsEncryptionContextForHv2(activeBranchKeyContext);
+
+    // get plaintext data key by calling kms::GenerateDataKey
+    var branchKey :- KMSKeystoreOperations.GetPlaintextDataKeyViaGenerateDataKey(
+      branchKeyContext := ecToKMS,
+      kmsConfiguration := kmsConfiguration,
+      keyManagerAndStorage := keyManagerAndStorage
+    );
+
+    // get beacon key by calling kms::GenerateDataKey
+    var beaconKey :- KMSKeystoreOperations.GetPlaintextDataKeyViaGenerateDataKey(
```

Comment:
Blocking/Suggestion:
I encourage labeling these "plaintext" as it is a meaningful distinction b/w HV-1 and HV-2 that the plain-text material is exposed to Admin agents.
I encourage labeling these "material" as we have so many "keys", "cipher-texts", and "plain-texts" in our specification and libraries.
I insist on using names/labels that are unique.
I recognize that this appears to be a minor thing,
and that my strong position may seem to be out of proportion to it.

But, I have had to explain these concepts to many people too many times to not have a strong opinion.  
```suggestion
    var beaconPlaintextMaterial :- KMSKeystoreOperations.GetPlaintextDataKeyViaGenerateDataKey(
```